Update templates to use file scoped namespace declarations

DamianEdwards opened the pull request:
Depends on dotnet/roslyn#54763 being merged and flowing to SDK before dotnet/aspnetcore can get an SDK with support for this feature in the compiler.

Contributes to #33947 #33948

DamianEdwards committed:
Update templates to use file-scoped namespace declarations

Contributes to #33947

DamianEdwards committed:
Update SPA templates to use file-scoped namesapces

#33947

DamianEdwards committed:
Update Razor page template to use file-scoped namespaces

#33947

pranavkm commented on src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Controllers/WeatherForecastController.cs:
```diff
@@ -1,128 +1,120 @@
+    {
+        HttpContext.VerifyUserHasAnyAcceptedScope(scopeRequiredByApi);
+
+        using var response = await _downstreamWebApi.CallWebApiForUserAsync("DownstreamApi").ConfigureAwait(false);
```
@davidfowl does this trigger you?

DamianEdwards replied:
For the record, I didn't change that, that's simply what was there already 🙃

davidfowl replied:
What the *&^%%$#@ is this

davidfowl replied:
Why is it using configure await, I have so many questions.

JunTaoLuo replied:
This was part of the Identity.Web work: https://github.com/dotnet/aspnetcore/issues/24024

pranavkm reviewed (approved):
As long as it compiles!

DamianEdwards commented:
/azp run

DamianEdwards commented:
So turns out the right branch of Roslyn wasn't flowing into the SDK so we would've been waiting a while for the feature to appear. That's now [been fixed](https://github.com/dotnet/sdk/commit/6dd2472dd83e6b8aae2947391ee3a59fe66934c2) so once code flow results in a new SDK being produced and flowed to installer which in turn flows to /aspnetcore (right?) we can finally get this change in.

DamianEdwards committed:
Move to SDK version that supports file-scoped namespaces

dougbu commented:
@DamianEdwards this is going to conflict w/ the move to https://github.com/dotnet/spa-templates and my current work on the second phase of #32027 (using the new repo as a submodule in this one). Any chance this could wait a day so that your changes to the Spa template could be made in the new repo❔

DamianEdwards commented:
@DamianEdwards we're trying to get this in for preview.7, is your change targeting preview.7 too?

dougbu commented:
> we're trying to get this in for preview.7, is your change targeting preview.7 too?

Yes❕ I'll put up the "phase 2" PR shortly after lunch I hope.

DamianEdwards commented:
@dougbu so should I just undo the changes to the SPA templates given they're moving?

dougbu commented:
> @dougbu so should I just undo the changes to the SPA templates given they're moving?

That'll work for me if it works for you.

DamianEdwards committed:
Revert "Update SPA templates to use file-scoped namesapces"

This reverts commit 97ff3daf093189a25259caab2d839a789806c0ca.

DamianEdwards committed:
Make CA1305 a suggestion rather than error

New compiler now makes this appear when using a StringBuilder with string interpolation. I'll log an issue to follow up.

DamianEdwards commented:
Disabled the warnings that started showing up with the new SDK build (thanks to the new interpolated string builders language feature) on advice from @pranavkm.
